app/src/components/templateEditor/TemplateEditor.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useMemo } from "react";
+import { nord } from "@uiw/codemirror-theme-nord";
+import CodeMirror, { ReactCodeMirrorProps } from "@uiw/react-codemirror";
+
+import { useTheme } from "@phoenix/contexts";
+import { assertUnreachable } from "@phoenix/typeUtils";
+
+import { FStringTemplating } from "./language/fStringTemplating";
+import { MustacheLikeTemplating } from "./language/mustacheLikeTemplating";
+
+export const TemplateLanguages = {
+  FString: "f-string", // {variable}
+  Mustache: "mustache", // {{variable}}
+} as const;
+
+type TemplateLanguage =
+  (typeof TemplateLanguages)[keyof typeof TemplateLanguages];
+
+type TemplateEditorProps = ReactCodeMirrorProps & {
+  templateLanguage: TemplateLanguage;
+};
+
+export const TemplateEditor = ({
+  templateLanguage,
+  ...props
+}: TemplateEditorProps) => {
+  const { theme } = useTheme();
+  const codeMirrorTheme = theme === "light" ? undefined : nord;
+  const extensions = useMemo(() => {
+    const ext: TemplateEditorProps["extensions"] = [];
+    switch (templateLanguage) {
+      case TemplateLanguages.FString:
+        ext.push(FStringTemplating());
+        break;
+      case TemplateLanguages.Mustache:
+        ext.push(MustacheLikeTemplating());
+        break;
+      default:
+        assertUnreachable(templateLanguage);
+    }
+    return ext;
+  }, [templateLanguage]);
+
+  return (
+    <CodeMirror theme={codeMirrorTheme} extensions={extensions} {...props} />
```

Comment:
I see the ...props so I get that you are trying to keep this generic. Maybe you can expose under `/code` a version that doesn't have things like line numbers since I think that's the primary use-case

Reply:
I'm happy to tweak any defaults you want for TemplateEditor, including removing line numbers and setting a default height.

I tend to opt for exposing all props on components so that I don't have to add them later but I'm happy with leaving extra props opaque until they're actually needed to be drilled through

Reply:
<img width="922" alt="image" src="https://github.com/user-attachments/assets/28460f7d-2e09-4ec5-8eae-0d3a43286cfa">

No more gutter, line numbers, or line highlight by default